### PR TITLE
fix: improve logs messages and error handling

### PR DESCRIPTION
**Pull request by jooola**
- Refactor the error handling using a helper function.
- Improve the logs messages and error messages.

### CODE REVIEW

File: builder/hcloud/helper.go [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+	"fmt"
+
+	"github.com/hashicorp/packer-plugin-sdk/multistep"
+	packersdk "github.com/hashicorp/packer-plugin-sdk/packer"
+)
+
+// errorHandler is a helper function to reduce the amount of bloat and complexity
+// caused by redundant error handling logic.
+func errorHandler(state multistep.StateBag, ui packersdk.Ui, prefix string, err error) multistep.StepAction {
+	wrappedError := err
```
Comment (phm07) [inline builder/hcloud/helper.go:13]: Why do we need to introduce a new variable here?

File: builder/hcloud/step_create_server.go [hunk 2]
```diff
@@ -44,13 +43,10 @@ func (s *stepCreateServer) Run(ctx context.Context, state multistep.StateBag) mu
 	for _, k := range c.SSHKeys {
 		sshKey, _, err := client.SSHKey.Get(ctx, k)
 		if err != nil {
-			ui.Error(err.Error())
-			state.Put("error", fmt.Errorf("Error fetching SSH key: %s", err))
-			return multistep.ActionHalt
+			return errorHandler(state, ui, "Could not fetch SSH key", err)
```
Comment (pauhull) [inline builder/hcloud/step_create_server.go:46]: Why is `k` omitted from the error message?

File: builder/hcloud/helper.go (hunk 1 above)
Comment (jooola) [reply builder/hcloud/helper.go:13]: I find it more explicit.

File: builder/hcloud/step_create_server.go [hunk 3]
```diff
@@ -44,13 +43,10 @@ func (s *stepCreateServer) Run(ctx context.Context, state multistep.StateBag) mu
...
 		}
 		if sshKey == nil {
-			state.Put("error", fmt.Errorf("Could not find key: %s", k))
-			return multistep.ActionHalt
+			return errorHandler(state, ui, "Could not find SSH key", err)
```
Comment (phm07) [inline builder/hcloud/step_create_server.go]: Why is `k` omitted from the error message?

**Review by phm07 — APPROVED** [review]

**Pull request merged**